apiserver/facades/client/application/application.go

```diff
@@ -1680,7 +1678,7 @@ func (api *APIBase) DestroyConsumedApplications(ctx context.Context, args params
 
 // ScaleApplications scales the specified application to the requested number of units.
 func (api *APIBase) ScaleApplications(ctx context.Context, args params.ScaleApplicationsParams) (params.ScaleApplicationResults, error) {
-	if api.model.Type() != state.ModelTypeCAAS {
+	if api.modelInfo.Type == model.IAAS {
```

Comment:
I would rather keep negating against CAAS as we types of models could be more then 2 and this check would still work.